allennlp/interpret/influence_interpreters/influence_interpreter.py

```diff
@@ -0,0 +1,361 @@
+        test_data_loader.index_with(self._vocab)
+        instances = list(test_data_loader.iter_instances())
+        return self.interpret_instances(instances, k=k)
+
+    def interpret_instances(
+        self, test_instances: List[Instance], k: int = 20
+    ) -> List[InterpretOutput]:
+        """
+        Run the influence function scorer on the given instances, returning the top `k`
+        most influential train instances and the bottom `k` least influential train instances
+        for each test instance.
+
+        !!! Note
+            Test instances should have `targets` so that a loss can be computed.
+        """
+        outputs: List[InterpretOutput] = []
+        for test_idx, test_instance in enumerate(Tqdm.tqdm(test_instances, desc="test instances")):
+            test_batch = Batch([test_instance])
+            test_batch.index_instances(self._vocab)
+            test_tensor_dict = move_to_device(test_batch.as_tensor_dict(), self._device)
+
+            # Prepare model for loss and gradient calculations.
+            self._model.eval()
+            self._model.zero_grad()
+
+            # Compute loss with respect to the test instance.
+            test_output_dict = self._model(**test_tensor_dict)
+            test_loss = test_output_dict["loss"]
+            test_loss_float = test_loss.detach().item()
+
+            # Get the (parameter) gradients with respect to the test loss.
+            test_grads = autograd.grad(test_loss, self._used_params)
+            # Sanity check.
+            assert len(test_grads) == len(self._used_params)
+
+            # Get influence scores.
+            influence_scores = torch.zeros(len(self.train_instances))
+            for idx, score in enumerate(
+                self.calculate_influence_scores(test_instance, test_loss_float, test_grads)
+            ):
+                influence_scores[idx] = score
+
+            # Gather top k and bottom k.
+            top_k_scores, top_k_indices = torch.topk(influence_scores, k)
+            bottom_k_scores, bottom_k_indices = torch.topk(-influence_scores, k)
+            top_k = self._gather_instances(top_k_scores, top_k_indices)
+            bottom_k = self._gather_instances(bottom_k_scores, bottom_k_indices)
+
+            outputs.append(
+                InterpretOutput(
+                    test_instance=test_instance,
+                    loss=test_loss_float,
+                    top_k=top_k,
+                    bottom_k=bottom_k,
+                )
+            )
+        return outputs
+
+    def _gather_instances(
+        self, scores: torch.Tensor, indices: torch.Tensor
+    ) -> List[TrainInstanceInfluence]:
+        outputs: List[TrainInstanceInfluence] = []
+        for score, idx in zip(scores, indices):
+            instance, loss, _ = self.train_instances[idx]
+            outputs.append(TrainInstanceInfluence(instance=instance, loss=loss, score=score.item()))
+        return outputs
+
+    def calculate_influence_scores(
```

Comment:
This looks like it should be private?  I get that you want people writing new `InfluenceInterpreters` to see this and know they have to implement this method, but you don't want people _using_ an `InfluenceInterpreter` to call this method, or even have to think about it.  Having it here with a public name makes me as a user spend time wondering if I should be using this.

Reply:
Yea, I mainly wanted to make sure this showed up in the docs, but I forgot we actually have a way of allowing certain private methods to appear in the docs (like we do with `DatasetReader._read()`). I'll update to make this private and include in the docs.